wpilibc/athena/src/DriverStation.cpp

```diff
@@ -584,9 +582,13 @@ void DriverStation::Run() {
   while (m_isRunning) {
     HAL_WaitForDSData();
     GetData();
+    // notify IsNewControlData variables
+    m_newControlData = true;
+
+    // notify WaitForData block
     {
       std::lock_guard<priority_mutex> lock(m_waitForDataMutex);
-      m_updatedControlLoopData = true;
+      m_waitForDataPredicate = true;
     }
     m_waitForDataCond.notify_all();
```

Comment:
I've always seen in C++ people calling the notify while not locked in the mutex. I know Java and other languages don't allow that, but most of the C++ code I've seen does this.